CMSgov/mint-app
# [EASI-3610] Update gainshare eligibility question

# EASI-3610

## Changes and Description

Fixes bug where export modal doesnt open on the view view

Backend Changes:

- Updated gql schema
- Created flyway migration
- Updated data model
- Updated unit test
- Updated postman collection

Frontend Changes:

- Updated GQL queries to include new fields 
  - `gainsharePaymentsEligibility`
  - `gainsharePaymentsEligibilityOther`
- Updated forms to include new condition question
- Updated readonly to render new data
- Updated CSV to include new fields
- Updated unit tests
- Updated cypress tests



## How to test this change

Backend Instructions:

- Start containers
- Verify plan participants and providers have new columns gainsharePaymentsEligibility and gainsharePaymentsEligibilityOther
- Run Postman
- Verify you can set appropriate values for the corresponding columns

Frontend Instructions:

- Log in and navigate to page 4 (Coordination) of Participants and Providers
- Click `Yes` to gainsharing payments question
- Verify new question is listed as a conditional followup - `Are providers eligible to participate in gainsharing arrangements?`
- Verify if selecting `Other` that the conditional text field appears
- Verify selections persist and render on the form upon reentry
- Verify data is rendered in readonly view
- Verify data is rendered in CSV export

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
